packages/@react-spectrum/tabs/src/Tabs.tsx

```diff
@@ -121,40 +193,157 @@ export function Tab<T>(props: TabProps<T>) {
   let [style, setStyle] = useState({
     width: undefined,
     height: undefined,
-    transform: undefined
+    transform: `translateX(var(--spectrum-tabs-focus-ring-padding-x))`
   });
 
   useLayoutEffect(() => {
-    let styleObj = {transform: undefined, width: undefined, height: undefined};
-    styleObj.transform = orientation === 'vertical'
-      ? `translateY(${selectedTab.offsetTop + verticalSelectionIndicatorOffset / 2}px)`
-      : `translateX(${selectedTab.offsetLeft}px) `;
-
-    if (orientation === 'horizontal') {
-      styleObj.width = `${selectedTab.offsetWidth}px`;
-    } else {
-      styleObj.height = `${selectedTab.offsetHeight - verticalSelectionIndicatorOffset}px`;
+    if (selectedTab) {
+      let styleObj = {transform: undefined, width: undefined, height: undefined};
+      // In RTL, calculate the transform from the right edge of the tablist so that resizing the window doesn't break the Tabline position due to offsetLeft changes
+      let offset = direction === 'rtl' ? -1 * (selectedTab.offsetParent?.offsetWidth - selectedTab.offsetWidth - selectedTab.offsetLeft) : selectedTab.offsetLeft;
+      styleObj.transform = orientation === 'vertical'
+        ? `translateY(${selectedTab.offsetTop + verticalSelectionIndicatorOffset / 2}px)`
+        : `translateX(${offset}px)`;
+
+      if (orientation === 'horizontal') {
+        styleObj.width = `${selectedTab.offsetWidth}px`;
+      } else {
+        styleObj.height = `${selectedTab.offsetHeight - verticalSelectionIndicatorOffset}px`;
+      }
+      setStyle(styleObj);
     }
-    setStyle(styleObj);
-  }, [direction, setStyle, selectedTab, orientation, scale]);
+
+  }, [direction, setStyle, selectedTab, orientation, scale, verticalSelectionIndicatorOffset, selectedKey]);
 
   return <div className={classNames(styles, 'spectrum-Tabs-selectionIndicator')} role="presentation" style={style} />;
 }
+
+interface TabListProps<T> {
+  isQuiet?: boolean,
+  density?: 'compact',
+  isDisabled?: boolean,
+  orientation?: Orientation,
+  state: SingleSelectListState<T>,
+  selectedTab: HTMLElement
+}
+
+const TabList = React.forwardRef(function <T> (props: TabListProps<T>, ref) {
+  let {
+    isQuiet,
+    density,
+    state,
+    isDisabled,
+    orientation,
+    selectedTab,
+    ...otherProps
+  } = props;
+
+  return (
+    <div
+      {...otherProps}
+      ref={ref}
+      className={classNames(
+        styles,
+        'spectrum-Tabs',
+        `spectrum-Tabs--${orientation}`,
+        {
+          'spectrum-Tabs--quiet': isQuiet
+        },
+        density ? `spectrum-Tabs--${density}` : '',
+        'spectrum-Tabs--container'
+      )}>
+      {[...state.collection].map((item) => (
+        <Tab key={item.key} item={item} state={state} isDisabled={isDisabled} orientation={orientation} />
+      ))}
+      <TabLine orientation={orientation} selectedTab={selectedTab} />
+    </div>
+  );
+});
+
+function TabsPicker(props) {
```

Comment:
Why is tabs plural? TabsPicker instead of TabPicker. I'm comparing this to TabList.